### PR TITLE
Add mit scene parsing benchmark loader

### PR DESCRIPTION
**Pull request by josephreisinger**
I haven't worked out how to convert the raw ADE20k dataset to the MIT sceneparsing label set, but this loader works with the actual benchmark training dataset (150 classes) directly and trains fine.

Fetch data from:
```
wget http://data.csail.mit.edu/places/ADEchallenge/ADEChallengeData2016.zip
unzip ADEChallengeData2016.zip
```

Edit config to point to the data and run using:
```
python train.py --arch fcn8s --dataset mit_sceneparsing_benchmark
```

### CODE REVIEW

**Comment by meetshah1995** [comment]
Looks great! Thanks for the docstrings and sanity checks :). I'll propagate them to other dataloaders over the weekend.
